RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -56,4 +76,31 @@ private String convertToBulkRequestBody(List<String> bulkSections) {
         }
         return builder.toString();
     }
+
+    private static java.util.function.Predicate<String> bufferPredicate(int maxItems, long maxSizeInBytes) {
+        return new java.util.function.Predicate<>() {
+            private int currentItemCount = 0;
+            private long currentSize = 0;
+
+            @Override
+            public boolean test(String next) {
+                currentItemCount++;
+                // TODO: Move to Bytebufs to convert from string to bytes only once
+                // Add one for newline between bulk sections
+                currentSize += next.getBytes(StandardCharsets.UTF_8).length + 1;
+
+                // Return true to keep buffering while conditions are met
+                if (currentSize == 0 ||
+                    (currentItemCount <= maxItems && currentSize <= maxSizeInBytes)) {
+                    return true;
+                }
+
+                // Reset and return false to signal to stop buffering.
+                // Next item is excluded from current buffer
+                currentItemCount = 0;
+                currentSize = 0;
+                return false;
```

Comment:
There's an annotation here that `Added lines #L100 - L102 were not covered by tests` which surprises me -- can we add a test that exercises finding the end of a bulk request (ideally one by size and one by length)? (I assume our tests have 1000 docs and none of our test snapshots have that many documents)

Reply:
Covered this thoroughly in tests now